### DISCUSSION TITLE
How to reset the internal ID counters for unit testing

### DISCUSSION THREAD

**Discussion by cmlenz** [tailwindlabs/headlessui#1287] [Help]
Does anyone know of a way to reset the internal ID counter that Headless UI uses to generate unique IDs, and the ARIA attributes based on those IDs? In unit testing, especially snapshot testing, these IDs often change depending on which tests ran before, so I'm looking for a way to reset the counter in a `beforeEach`.

**Comment by arlyon** [comment]
I have a PR for this here: https://github.com/tailwindlabs/headlessui/pull/1776
